config/database.yml

```
@@ -49,8 +49,24 @@ test:
 # Read https://guides.rubyonrails.org/configuring.html#configuring-a-database
 # for a full overview on how database connection configuration can be specified.
 #
+
+
+# ローカル環境でもこのファイル全体が読み込まれる (評価される)
+# development 環境の credentials は存在しておらず参照エラーになるので、ぼっち演算子を使用
 production:
   <<: *default
-  database: app_production
-  username: app
-  password: <%= ENV["APP_DATABASE_PASSWORD"] %>
+  database: <%= Rails.application.credentials.db&.database %>
+  username: <%= Rails.application.credentials.db&.username %>
+  host: <%= Rails.application.credentials.db&.host %>
+  password: <%= Rails.application.credentials.db&.password %>
+  ssl_mode: <%= Rails.application.credentials.db&.ssl_mode %>
+  sslca: <%= Rails.application.credentials.db&.sslca %>
```

Rails.env.production?でローカルか本番かを切り分けできるのでこれを入れておくのが良いかもです。
stagingも同様です。

```suggestion
  <% if Rails.env.production? %>
    database: <%= Rails.application.credentials.db.database %>
    username: <%= Rails.application.credentials.db.username %>
    host: <%= Rails.application.credentials.db.host %>
    password: <%= Rails.application.credentials.db.password %>
    ssl_mode: <%= Rails.application.credentials.db.ssl_mode %>
    sslca: <%= Rails.application.credentials.db.sslca %>
  <% end %>
```

---

基本的に`&.`などは設定周りではほとんど使うケースがない(nilが入ることを前提としたコードを書くべきではない)という感じなので使わない方が良いと思います。

アプリケーション側のコードでは結構使うので全然良いと思いますが。

アプリケーション側も事前にnil対策できるものについては基本的にはnilを前提としたコードにしない方が良いというのはありますね。

---

@stkzk3110 
教えていただきありがとうございます！  
yaml ファイルにて、各項目の設定値の記述で erb (?) を使う頭はあったのですが、複数行を囲むのは意識になかったです。  
view ファイルで考えたら当たり前のことなのですが、なぜだか ":" の後ろに使うものという感じの頭になっていました。

`&.` を使わなくてもローカル環境で動作させる方法として config/credentials/development.yml.enc を設置し定義だけしておく、という方法を思いついていたのですが、  
なんだかピンと来ないやり方だったのでレビューに回してよかったです。

教えていただいた記述方法で production, staging あたりのデータベース設定をしてみます。

---

[bc6421b](https://github.com/commew/timelogger-api/pull/26/commits/bc6421b9bff7d32d28f8292db3fab73ab2e2e7bd)
[e763788](https://github.com/commew/timelogger-api/pull/26/commits/e763788378d7ceea94aeacda9f630b1d47bf107e)
にて、環境ごとの if 文を追加する修正を行いました。

その状態で動作確認を行ったところインデントエラーが発生したため、  
以下のコミットにてインデントの調整と if 文の記述位置の調整を行いました。
[3ea0e28](https://github.com/commew/timelogger-api/pull/26/commits/3ea0e28d17e79a88b2613bcaad746b5e3f7b1eb1)

やり取りが増えてしまったため、最終的なコードのリンクも添付いたします。
https://github.com/commew/timelogger-api/blob/3ea0e28d17e79a88b2613bcaad746b5e3f7b1eb1/config/database.yml

@stkzk3110 
追加で行った修正につきまして問題などがなさそうか、レビューをお願いできますでしょうか？

---

@HaruyaFujimoto 
遅くなってすみません。
確認しました〜

Approveです！